arduino-ide-extension/src/node/core-service-impl.ts

```diff
@@ -304,9 +345,36 @@ export class CoreServiceImpl extends CoreClientAware implements CoreService {
     }
   }
 
-  private flushOutputPanelMessages(chunk: string): void {
-    this.responseService.appendToOutput({
-      chunk,
-    });
+  private createPort(port: Port | undefined): GrpcPort {
+    const grpcPort = new GrpcPort();
+    if (port) {
+      grpcPort.setAddress(port.address);
+      grpcPort.setLabel(port.addressLabel);
+      grpcPort.setProtocol(port.protocol);
+      grpcPort.setProtocolLabel(port.protocolLabel);
+    }
+    return grpcPort;
+  }
+}
+/**
+ * Artificial common interface for all gRPC streaming requests.
+ * Such as `UploadResponse,` `UploadUsingProgrammerResponse`, `BurnBootloaderResponse`, and the `CompileResponse`.
+ */
+interface StreamingResponse {
```

Comment:
Why not union types?

Reply:
https://github.com/arduino/arduino-ide/pull/1064/commits/c8e22407db5c24d8b11ee0fd16c6cf4b1b8441c2